src/sql/sqlops.d.ts

```diff
@@ -209,6 +209,31 @@ declare module 'sqlops' {
 		id: string;
 	}
 
+	/**
+ 	* Options for the actions that could happen after connecting is complete
+ 	*/
+	export interface IConnectionCompletionOptions {
```

Comment:
I think everything other than `saveConnection` should be optional properties to minimize the work of anyone using this API.